docs/custom_detect_languages_func.rst

```diff
@@ -0,0 +1,65 @@
+============
+Custom language detection
+============
+
+Usage
+=====
+:func:`dateparser.parse` supports two language detection
+libraries `fasttext <https://github.com/facebookresearch/fastText>`_ 
+and `langdetect <https://github.com/Mimino666/langdetect>`_.
+
+fastText
+~~~~~~~~
+Language detection with fastText.
+
+Import fasttext wrapper and pass it as ``detect_languages_func``
+parameter with::
+
+    >>> from dateparser.custom_lang_detect.fasttext import detect_languages
+    >>> dateparser.parse('12/12/12', detect_languages_func=detect_languages)
+
+fastText supports currently supports fastText - large and small model you can
+download your model of choice using 
+
+`Read more <https://fasttext.cc/blog/2017/10/02/blog-post.html>`_ about models.
+
+
+langdetect
+~~~~~~~~
+Language detection with langdetect.
+
+Import langdetect wrapper and pass it as ``detect_languages_func``
+parameter with::
+
+    >>> from dateparser.custom_lang_detect.langdetect import detect_languages
+    >>> dateparser.parse('12/12/12', detect_languages_func=detect_languages)
+
+
+.. note::
+
+    ``fastText`` is the fastest if faster,  more 
+    accurate and supports more languages.
+
+Custom implementation
+=====================
+
+dateparser allows the integration of any library
+you can implement any language detection library by wrapping it with a 
+function accepting ``text`` and ``confidence_threshold`` and returning
+list of detected language codes.
+
+
+Wrapper for boilerplate for implementing custom language detections::
+
+    def detect_languages(text, confidence_threshold):
+        language_codes = []
+        parser_data = languages_dectetion_function(text)
+        for language_candidate in parser_data:
+            if language_candidate > confidence_threshold:
+                language_codes.append(language_code)
+        return language_codes
```

Comment:
I would suggest changing this to a description.

```suggestion
    def detect_languages(text, confidence_threshold):
        """
        Takes two variables: `text` and `confidence_treshold` and returns
        a list of `languages codes`.
        
        * `text` is a string containing from where the language codes are 
        derived.
        
        * `confidence_treshold` is a number between 0 and 1 that can be 
        used to decide if the confidence is enough. It can be also ignored.
        This value comes from the dateparser setting: 
        `LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD`
        
        The result must be a list of languages codes (strings).
        """
        # here you can apply your own logic
        return language_codes
```

cc: @lopuhin

Reply:
Sounds good to me, thanks @noviluni